spec/state_machines/allocation_state_machine_spec.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe AllocationStateMachine do
+  let(:machine) { described_class.new(target) }
+  let(:target) { Struct.new(:status).new(initial_state) }
+  let(:initial_state) { :unfilled }
+
+  before { machine.restore!(initial_state) }
+
+  shared_examples 'state_machine target status' do |expected_status|
+    describe 'machine status' do
+      it { expect(machine.current).to eql expected_status }
+    end
+
+    describe 'target status' do
+      it { expect(target.status).to eql expected_status }
+    end
+  end
+
+  it { is_expected.to respond_to(:fill, :unfill, :cancel) }
+
+  context 'when in the unfilled status' do
+    it_behaves_like 'state_machine target status', :unfilled
+
+    context 'when the fill event is fired' do
+      before { machine.fill }
+
+      it_behaves_like 'state_machine target status', :filled
+    end
+
+    context 'when the unfill event is fired' do
+      before { machine.unfill }
+
+      it_behaves_like 'state_machine target status', :unfilled
+    end
+
+    context 'when the cancel event is fired' do
+      before { machine.cancel }
+
+      it_behaves_like 'state_machine target status', :cancelled
+    end
+  end
+
+  context 'when in the filled status' do
+    let(:initial_state) { :filled }
+
+    it_behaves_like 'state_machine target status', :filled
+
+    context 'when the unfill event is fired' do
+      before { machine.unfill }
+
+      it_behaves_like 'state_machine target status', :unfilled
+    end
+
+    context 'when the fill event is fired' do
+      before { machine.fill }
+
+      it_behaves_like 'state_machine target status', :filled
+    end
+
+    context 'when the cancel event is fired' do
+      before { machine.cancel }
+
+      it_behaves_like 'state_machine target status', :cancelled
+    end
+  end
+
+  context 'when in the none status' do
```

Comment:
Is this the same as having a `nil` persisted state in the allocation? If so I would expect the line below to assign `:initial_state` to `nil` rather than `:none`. I'm not sure if `:none` is a magic value representing `nil` or would end up being a named state (which we don't really want).

Reply:
so on revisiting this is available by default and should not be specified this way, so I removed it